### DISCUSSION TITLE
Challenge 29 help!

### DISCUSSION THREAD

**Discussion by djvinnie** [OWASP/wrongsecrets#946] [Q&A] [answered]
I have been working my way through the challenges over the weekend and they have been super fun! With challenge 29 though, I am little confused - I did look through all issues with screenshots in them and [#742](https://github.com/OWASP/wrongsecrets/issues/742) stood out to me in particular but when I try the Generated Security Password as the answer, it is not working for me..! Am I missing anything here? Appreciate the help!

**Comment by commjoen** [comment]
Hi @djvinnie ! I hope to investigate the issue this week, looks like the decryption of the challenge fails resulting in the wrong answer. I hope to get back to you soon ;-)

**Reply by commjoen** [reply]
It seems to say in the logs; 
2023-08-27T05:29:07.276547+00:00 app[web.1]: 2023-08-27T05:29:07.276Z  WARN 3 --- [io-51037-exec-2] o.o.w.challenges.docker.Challenge29      : Exception when decrypting: src/test/resources/RSAprivatekey.pem

**Reply by djvinnie** [reply]
Appreciate it @commjoen ! While we are on this, can I also point out that the website for challenge 32 seems to be down and it does not look like it will come back up for a while. Could this be checked as well?

**Reply by commjoen** [reply]
https://gpa.43z.one/ takes some time to come up. It works now

**Reply by commjoen** [reply]
Re32: if you visit once and then 15 minute later it should work. Will create a pr to make that more clear. I can fix that later this/next week. Or feel free file a pr in the mean time.

Re29: it seems we forgot to make the path loading in https://github.com/OWASP/wrongsecrets/blob/c6f854582b327c856d8dd4968e6e173f392da316/src/main/java/org/owasp/wrongsecrets/challenges/docker/Challenge29.java#L77 dynamic. I can fix that later this/next week. Or feel free file a pr in the mean time.

**Reply by commjoen** [reply]
Re32: initial extra text to explain added in https://github.com/OWASP/wrongsecrets/pull/948

**Reply by djvinnie** [reply]
Re32: Thank you for the change!
Re29: I will find time this week to file a pr.

**Reply by commjoen** [reply]
That would be awesome @djvinnie ! I first have some PR-reviews todo before I can jump onto Re29 :) .

**Comment by commjoen** [comment]
https://github.com/OWASP/wrongsecrets/pull/949 could possible fix challenge29

**Reply by commjoen** [reply]
i will create a container out of it to test it :)

**Comment by commjoen** [comment]
Re29: it is now fixed in https://github.com/OWASP/wrongsecrets/pull/949 ;) . you can test this already locally by doing:

`docker run -p 8080:8080 jeroenwillemsen/wrongsecrets:1.6.9RC2-no-vault`
Now try to solve challenge 29 at http://localhost:8080/challenge/29 ;-)

**Reply by commjoen** [reply]
We will merge and release later :) if you see it working on your machine as well :).

**Reply by djvinnie** [reply]
Heyy you said you'll look into it later during the week! Thank you though, once I am done with all of the challenges I will start solving any arising issues in the application. I will test out the challenge fix in some time.

**Reply by djvinnie** [reply]
Re29: Worked perfectly fine!
Re32: I opened the hxxps://gpa[.]43z[.]one site, wrote my prompt and saw the 502 error. I waited for a couple of minutes and tried again, and I got a 503 (service temporarily unavailable) error. I then waited for 15 minutes, refreshed the website and tried my prompt again but it still did not work for me. 🤔

**Reply by commjoen** [reply]
Hmm it seems dead now:
HELP: Keep this game alive
Unfortunately this game could stop being playable at any moment. It's very costly to keep it running - peak times $15/day. If you want to keep it running for longer so everyone can enjoy it, please send a few bucks to my bitcoin address bc1quuwxf4a2tge96ydz7f5dck2cp7yrs8zkge4w3x if you can afford it. Or DM me an API key that you don't need anymore. Anyway It was fun while it lasted :)

**Reply by commjoen** [reply]
I guess we need to disable the challenge for now. What do you think @bendehaan ?

**Reply by commjoen** [reply]
Re32: we can replace it with https://gandalf.lakera.ai/

**Comment by commjoen** [comment, marked as answer]
Now replaced 32 with Gandalf: https://github.com/OWASP/wrongsecrets/pull/950 . And we fixed 29 with https://github.com/OWASP/wrongsecrets/pull/949 . This means both issues are addressed in this discussion. I will close this discussion to ensure we have some oversight on all the things.
Thank you for playing and validating the challenges!
Can you please, for any new issue you encounter open a new discussion/issue on github or contact us on slack :) ?

**Reply by djvinnie** [reply]
Thank you for the quick turnaround with the issues I encountered @commjoen !

Sure, I will be happy to start a new discussion thread here. Since you mentioned slack, how do I access the slack channels for the project? Is there a link that you can provide me?

**Reply by commjoen** [reply]
Need support? Contact us via [OWASP Slack](https://owasp.slack.com/archives/C02KQ7D9XHR) for which you sign up [here](https://owasp.org/slack/invite)
